### ISSUE TITLE
nvda does not read the button label in whatsapp when in browse mode 

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#12633]
### Steps to reproduce:
install whatsapp on your computer, 64bit version obtained from whatsapp website.
nvda is configured not to enter focus mode automatically.
the configuration
Automatically position the system focus on focusable elements
is disabled
open whatsapp, connect your phone.
choose a conversation and:
walk with tabe to the button
Open emoji panel button
change nvda to browse mode.
walk with the arrows below to:
attach button
press enter.
now walk with the arrows to up and notice some buttons being read only as button, the button labels are not read.
now put nvda in focus mode and continue to navigate the buttons with the arrows, see now that the labels are read. 
### Actual behavior:
when nvda is in browse mode the button labels are not read 
### Expected behavior:
button labels must be read in browse mode and focus mode 
#### NVDA installed/portable/running from source:
instaled
#### NVDA version:
nvda.exe, NVDA alpha-8,b8b88b0d

#### Windows version:
10.0.19043.1083
#### Name and version of other software in use when reproducing the issue:
WhatsApp.exe, WhatsApp 2.2123.8

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes
log:
[whatsapp.log](https://github.com/nvaccess/nvda/files/6794098/whatsapp.log)

**Comment by OzancanKaratas** [comment]
Please report this to WhatsApp Inc.

**Comment by fernando-jose-silva** [comment]
##- WhatsApp Support -##
Hi,
Thanks for your message.
Your experience is important to us. We're constantly working on improving our products. To stay updated with current and upcoming features, please visit our blog and Help Center frequently.
If you have any other questions or concerns, please feel free to contact us. We'll be happy to help!
 
good afternoon dears
I was using whatsapp desktop along with the nvda screen reader, and I noticed two problems:

When putting the nvda screen reader into focus mode and navigating with the tabe key it is not possible to put focus on various controls in the whatsapp interface.

Steps to reproduce:
install whatsapp from whatsapp website, 64bit
Connect your phone, make sure that nvda is in focus mode and walk with tabe through the whatsapp interface
note:
that you will go through:
Open emoji panel button
tab
blank editable multiline section
now return with shift + tabe to the button
Open emoji panel button
put nvda in browse mode and this with the arrows and note that you will go through several options not found when browsing with only tabe in focus mode:
Open emoji panel button
navigation mode
button Open GIF Panel
button Open the sticker panel
attach button
enter a message
editable multiline section
Actual behavior:
if you are in focus mode and walk with tabe you will find the options:
Open emoji panel button
tab
blank editable multiline section
Expected behavior:
if you are in focus mode and walk with tabe you will find the options:
Open emoji panel button
button Open GIF Panel
button Open the sticker panel
attach button
enter a message
editable multiline section
System configuration
NVDA installed/portable/running from source:
instaled
NVDA version:
nvda.exe, NVDA alpha-8,b8b88b0d
Windows version:
10.0.19043.1083
Name and version of other software in use when reproducing the issue:
WhatsApp.exe, WhatsApp 2.2123.8

I contacted the nvacces team that develops the nvda screen reader, and was asked to see with the whatsapp support team if the issues are caused by the whatsapp app.
follows the github ticket access link created for the nvda team.

https://github.com/nvaccess/nvda/issues/12634

again when using whatsapp together with the nvda screen reader i noticed another problem, when the screen reader is in hover mode it cannot read the label of some controls.
Again, when contacting the nvaccess team through github, it was suggested to check with the whatsapp team the possibility of the problem being with the whatsapp app.

Steps to reproduce:
install whatsapp on your computer, 64bit version obtained from whatsapp website.
nvda is configured not to enter focus mode automatically.
the configuration
Automatically position the system focus on focusable elements
is disabled
open whatsapp, connect your phone.
choose a conversation and:
walk with tabe to the button
Open emoji panel button
change nvda to browse mode.
walk with the arrows below to:
attach button
press enter.
now walk with the arrows to up and notice some buttons being read only as button, the button labels are not read.
now put nvda in focus mode and continue to navigate the buttons with the arrows, see now that the labels are read.
Actual behavior:
when nvda is in browse mode the button labels are not read
Expected behavior:
button labels must be read in browse mode and focus mode
System configuration
NVDA installed/portable/running from source:
instaled
NVDA version:
nvda.exe, NVDA alpha-8,b8b88b0d
Windows version:
10.0.19043.1083
Name and version of other software in use when reproducing the issue:
WhatsApp.exe, WhatsApp 2.2123.8

follow the github link for this second issue open to the nvaccess team

https://github.com/nvaccess/nvda/issues/12633

I would like to ask the whatsapp accessibility team to investigate these two issues and fix them if possible.
If the problem requires changes by nvaccess with the nvda screen reader, I would like to ask you to contact nvaccess in order to suggest the necessary fixes, I believe that whatsapp accessibility team has much more credibility to suggest changes .
I'm not a programmer.
The use of whatsapp in Brazil is very large.
Sorry for the flawed English, this is not my native language, these writings support the use of google translator.

--
Atenciosamente Fernando Silva.
request #61300297858570

**Comment by zersiax** [comment]
This is likely a problem with the markup used by WhatsApp. The same would happen with elements that have a title but not a label set, a tooltip ...all sorts of things

**Comment by fernando-jose-silva** [comment]
##- WhatsApp Support -##

Hi,

Sorry for the delay! We've received many emails recently. We do our best to answer them all. Thanks for your patience.

Thank you for reaching us and letting us know the different problems you are experiencing. I have escalated your feedback to our team to investigate the problems.

If you have any other questions or concerns, please feel free to contact us. We'll be happy to help!

Valentina 
WhatsApp Support Team

Take a look at our Help Center!

**Comment by leonardder** [comment]
Definitely an authoring error. The buttons are positioned in a list, and the list items get a label when they have focus.
